aiohttp_s3_client/client.py

```diff
@@ -691,3 +694,65 @@ async def get_file_parallel(
             with suppress(FileNotFoundError):
                 os.unlink(file_path)
             raise
+
+    async def list_objects_v2(
+        self,
+        object_name: t.Union[str, Path] = "/",
+        *,
+        bucket: t.Optional[str] = None,
+        prefix: t.Union[str, Path] = None,
+        delimiter: t.Optional[str] = None,
+        max_keys: t.Optional[int] = None,
+        start_after: t.Optional[str] = None,
+    ) -> t.AsyncIterator[t.List[AwsObjectMeta]]:
+        """
+        List objects in bucket.
+
+        Returns an iterator over lists of metadata objects, each corresponding
+        to an individual response result (typically limited to 1000 keys).
+
+        object_name:
+            path to listing endpoint, defaults to '/'; a `bucket` value is
+            prepended to this value if provided.
+        prefix:
+            limits the response to keys that begin with the specified
+            prefix
+        delimiter: a delimiter is a character you use to group keys
+        max_keys: maximum number of keys returned in the response
+        start_after: keys to start listing after
+        """
+
+        params = {
+            "list-type": "2",
+        }
+
+        if prefix:
```

Comment:
Just as recommendation:

```python
params = dict(
    prefix=str(prefix) if prefix else None,
    delimiter=delimiter or None,
    max_keys=str(max_keys) if max_keys else None,
    start_after=start_after or None,
    object_name=f"/{bucket}",
)
params = {k: v for k, v in params.items() if v is not None}
```

Reply:
To be honest I'm not sure it is an improvement as such; one could also use a list comprehension to create the `params` dict.

I think the code as it is now is acceptable.